spec/factories/segmented_user.rb

```
@@ -0,0 +1,6 @@
+FactoryBot.define do
+  factory :segmented_user do
+    user
+    audience_segment
+  end
+end
```

`Users::Delete`'s spec will go through all association will use factory bot to make sure each association are kept or destroy appropriately.

---

Should this comment live in the source for future readers to find?

---

Because the spec will fail loudly when this is missing, I've added the appropriate comment to the spec itself.